src/rucksack-1-freeimage.patch

```diff
@@ -0,0 +1,109 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 2fb544d..e5c09c7 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -19,12 +19,10 @@ else(C99_FLAG_DETECTED)
+ endif(C99_FLAG_DETECTED)
+ 
+ # check for freeimage
+-find_package(FreeImage)
+-if(FreeImage_FOUND)
+-  set(STATUS_FREEIMG "OK")
+-else(FreeImage_FOUND)
+-  set(STATUS_FREEIMG "not found")
+-endif(FreeImage_FOUND)
++if(NOT PKG_CONFIG_FOUND)
++  find_package(PkgConfig REQUIRED)
++endif()
++pkg_check_modules(FREEIMAGE freeimage)
+ 
+ # check for laxjson
+ find_package(LaxJson)
+@@ -72,7 +70,7 @@ set(EXE_HEADERS
+   ${PROJECT_SOURCE_DIR}/src/util.h
+   )
+ 
+-set(LIB_CFLAGS "${C99_C_FLAGS} -pedantic -Werror -Wall -Werror=strict-prototypes -Werror=old-style-definition -Werror=missing-prototypes -D_POSIX_C_SOURCE=200809L")
++set(LIB_CFLAGS "${C99_C_FLAGS} -pedantic -Werror -Wall -Werror=strict-prototypes -Werror=old-style-definition -Werror=missing-prototypes -D_POSIX_C_SOURCE=200809L ${FREEIMAGE_CFLAGS_OTHER}")
+ set(EXE_CFLAGS ${LIB_CFLAGS})
+ 
+ add_library(rucksack_static STATIC ${RUCKSACK_LIB_SOURCES} ${RUCKSACK_LIB_HEADERS})
+@@ -80,65 +78,30 @@ set_target_properties(rucksack_static PROPERTIES
+   OUTPUT_NAME rucksack
+   COMPILE_FLAGS ${LIB_CFLAGS})
+ 
+-add_library(rucksack_shared SHARED ${RUCKSACK_LIB_SOURCES} ${RUCKSACK_LIB_HEADERS})
+-set_target_properties(rucksack_shared PROPERTIES
+-  OUTPUT_NAME rucksack
+-  SOVERSION ${VERSION_MAJOR}
+-  VERSION ${VERSION}
+-  COMPILE_FLAGS ${LIB_CFLAGS})
+-
+ 
+-include_directories(${FreeImage_INCLUDE_DIRS})
++include_directories(${FREEIMAGE_INCLUDE_DIRS})
+ add_library(rucksackspritesheet_static STATIC
+   ${RUCKSACK_SPRITESHEET_LIB_SOURCES} ${RUCKSACK_SPRITESHEET_LIB_HEADERS})
+ set_target_properties(rucksackspritesheet_static PROPERTIES
+   OUTPUT_NAME rucksackspritesheet
+   COMPILE_FLAGS ${LIB_CFLAGS})
+ 
+-add_library(rucksackspritesheet_shared SHARED
+-  ${RUCKSACK_SPRITESHEET_LIB_SOURCES} ${RUCKSACK_SPRITESHEET_LIB_HEADERS})
+-set_target_properties(rucksackspritesheet_shared PROPERTIES
+-  OUTPUT_NAME rucksackspritesheet
+-  SOVERSION ${VERSION_MAJOR}
+-  VERSION ${VERSION}
+-  COMPILE_FLAGS ${LIB_CFLAGS})
+-target_link_libraries(rucksackspritesheet_shared ${FreeImage_LIBRARIES})
+-
+-
+ 
+ add_executable(rucksack ${EXE_SOURCES} ${EXE_HEADERS})
+-target_link_libraries(rucksack rucksack_shared rucksackspritesheet_shared ${LAXJSON_LIBRARY})
++target_link_libraries(rucksack rucksack_static rucksackspritesheet_static ${LAXJSON_LIBRARY} ${FREEIMAGE_LIBRARIES})
+ include_directories(${LAXJSON_INCLUDE_DIR})
+ set_target_properties(rucksack PROPERTIES
+   COMPILE_FLAGS ${EXE_CFLAGS})
+ 
+ install(TARGETS rucksack DESTINATION bin)
+ install(TARGETS
+-  rucksack_static rucksack_shared
+-  rucksackspritesheet_static rucksackspritesheet_shared
```

Comment:
I tried that first, and ran into a bunch of errors I couldn't figure out. Happy to try again if you're willing to help troubleshoot.